env/dev/r3broot/spack.yaml

```diff
@@ -1,26 +1,31 @@
 spack:
   definitions:
-    - when: platform == 'linux'
-      root: [root@6.16.00+fortran+gdml+memstat+pythia6+pythia8+vc~vdt+python+tmva+xrootd+sqlite ^mesa~llvm]
-    - when: platform == 'darwin'
-      root: [root@6.16.00+fortran+gdml+memstat+pythia6+pythia8+vc~vdt+python+tmva+xrootd+sqlite+aqua]
+  - when: platform == 'linux'
+    root: [root@6.16.00+fortran+gdml+memstat+pythia6+pythia8+vc~vdt+python+tmva+xrootd+sqlite
+        ^mesa~llvm]
+  - when: platform == 'darwin'
+    root: [root@6.16.00+fortran+gdml+memstat+pythia6+pythia8+vc~vdt+python+tmva+xrootd+sqlite+aqua]
+  - concretizer_hints:
+    - python@2.7.18
+    - py-numpy@1.16.6
+    - py-setuptools@44.1.0
   specs:
-    - python@2.7.16
-    - py-numpy@1.16.5
-    - googletest@1.8.1
-    - boost@1.68.0
-    - fairlogger@1.4.0
-    - dds@2.4
-    - fairmq@1.4.3
-    - pythia6@428-alice1
-    - hepmc@2.06.09 length=CM momentum=GEV
-    - pythia8@8240
-    - geant4@10.05.p01~qt~vecgeom~opengl~x11~motif~data~clhep~threads
-    - $root
-    - geant3@2-7_fairsoft
-    - vgm@4-5
-    - geant4-vmc@4-0-p1
-    - fairroot@18.2.1+sim
-    - r3broot
+  - $concretizer_hints
+  - fairsoft-config@jun19
```

Comment:
As the `r3broot` env is mainly for CI (maybe should be moved to `test/`?), why do you need to add fairsoft-config? Or does the old fairoot-Version (18.2) now really need it?

Reply:
> why do you need to add fairsoft-config?

FairRoot depends on it, I am just pinning the version to `jun19`.

Reply:
I am not sure, I like the "fairroot depends_on fairsoft-config".
* On the one hand, I can understand it.
* On the other hand: fairoot itself doesn't need it anymore, correct? So only the envs for expirements need it.

Reply:
(Technically it does not need to depend on it, but if it is not, I cannot use `spack view --dependencies true -e fairroot dir... fairroot`.)

Reply:
> As the r3broot env is mainly for CI (maybe should be moved to test/?

Sure, I don't mind where it is.

Reply:
> (Technically it does not need to depend on it, but if it is not, I cannot use `spack view --dependencies true -e fairroot dir... fairroot`.)

Or, is there another way to attach a view to an env and exclude `fairroot` without modifying the env?

Reply:
> > (Technically it does not need to depend on it, but if it is not, I cannot use `spack view --dependencies true -e fairroot dir... fairroot`.)
> 
> Or, is there another way to attach a view to an env and exclude `fairroot` without modifying the env?

I don't know, if this is an option, but what about adding the (relevant) packages as `depends_on()` to fairsoft-config? Like "If you install fairsoft-config, you expect the legacy fairsoft packages to be installed somewhere"?

Reply:
> but what about adding the (relevant) packages as depends_on() to fairsoft-config?

Because it will be exactly === all the run-time dependencies of the fairroot pkg.

Reply:
we concluded to keep it as dependency of fairroot offline.